torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1475,6 +1488,43 @@ def solve(self, disable_congruences=True, disable_equivalences=True):
         for source, expr in self._symbolic_equivalences:
             self._dynamic_results.add(f"{self._dcp.print_source(source)} == {self._dcp.doprint(expr)}")
 
+    @classmethod
+    def is_supported_equivalence(cls, expr):
+        # Currently supported Dim ops are linear expressions with integer coefficients.
+        # So check that expr only contains +, *, ints, and a single occurrence of a symbol
+        if isinstance(expr, (sympy.Add, sympy.Mul)):
+            lhs, rhs = expr.args
+            return (
+                (cls.is_supported_equivalence(lhs) and isinstance(rhs, sympy.Integer)) or
+                (isinstance(lhs, sympy.Integer) and cls.is_supported_equivalence(rhs))
+            )
+        return isinstance(expr, sympy.Symbol)
+
+    @classmethod
+    def solve_supported_equivalence(cls, expr, solution):
+        # Given a supported equivalence, solve it explicitly by inverting Dim ops.
+        # TODO(avik): maybe do this with sympy in the future
```

Comment:
@lezcano I feel like we've got another copy of this logic lying around somewhere?

Reply:
Yep, this is a simpler version fo `try_solve`.

Reply:
Hah yeah that's what jumped out to me today. (I'm sure `try_solve` does more.)

Reply:
See above. Btw I was worried a bit about the limit of trials in try_solve but I realized that because we have linear expressions (of the form Ax + B) we only really ever need 2 trials, and the current limit is 5.